### PR TITLE
Add email notify setting for downloads

### LINKED ISSUE #42: Add email notify setting for downloads
The repository module has settings to send email notifications on various events. It would be nice to add one setting for the download event.

### PR DESCRIPTION
**Pull request by juvander**
### Description of PR...
Added capability to send notifications into selected e-mail address when someone downloads file

## Changes made
Enabled e-mail sending and settings to control this. By default this is disabled

Closes #42

### CODE REVIEW

File: Settings.ascx.cs [hunk 1]
```diff
@@ -379,7 +381,17 @@ private void Page_Load(System.Object sender, System.EventArgs e)
 						cbxEmailOnComment.Checked = false;
 					}
 
-					if (!string.IsNullOrEmpty(Convert.ToString(settings["AnonEditDelete"]))) {
+
+                    if (!string.IsNullOrEmpty(Convert.ToString(settings["EmailOnDownload"])))
```
[the remark this answers is not in the document]
Comment (juvander) [reply Settings.ascx.cs]: True, will see these later today. Mostly this was just made same way as for other settings so they should be gone through as well similar way.

File: Repository.ascx.cs [hunk 2]
```diff
@@ -406,8 +407,47 @@ public void lstObjects_PageIndexChanged(object source, DataGridPageChangedEventA
...
+        {
+            try
+            {
+                // check to see if we need to send an email notification
+                if (string.IsNullOrEmpty(Convert.ToString(Settings["EmailOnDownload"])) == false)
+                {
+                    if (bool.Parse(Settings["EmailOnDownload"].ToString()) == true)
+                    {
+                        string _email = Convert.ToString(Settings["EmailOnDownloadAddress"]);
+                        if (string.IsNullOrEmpty(_email) == false)
```
[the remark this answers is not in the document]
Comment (juvander) [reply Repository.ascx.cs]: Matter of opinion, makes not so intuitive to read and end result is the same

File: Settings.ascx.cs [hunk 3]
```diff
@@ -392,7 +404,17 @@ private void Page_Load(System.Object sender, System.EventArgs e)
 						// default value
 					}
 
-					if (!string.IsNullOrEmpty(Convert.ToString(settings["EmailOnUpload"]))) {
+                    if (!string.IsNullOrEmpty(Convert.ToString(settings["EmailOnDownloadAddress"])))
```
[the remark this answers is not in the document]
Comment (juvander) [reply Settings.ascx.cs]: Convert.ToString does not throw an exception

**Comment by valadas** [comment]
Yep, the rest is more styling preferences, looks good to me.

**Pull request merged**